### PR TITLE
feat(DataArts): add dataarts data standard template resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dataarts data standard template resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
  add dataarts data standard template resource
**Release note**:

```release-note
  add dataarts data standard template resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dataarts/ TESTARGS='-run TestAccDataStandardTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts/ -v -run TestAccDataStandardTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccDataStandardTemplate_basic
=== PAUSE TestAccDataStandardTemplate_basic
=== CONT  TestAccDataStandardTemplate_basic
--- PASS: TestAccDataStandardTemplate_basic (14.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  14.860s
```

### CODE REVIEW

File: docs/resources/dataarts_architecture_data_standard_template.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `workspace_id` - (Required, String, ForceNew) Specifies the workspace ID of DataArts Architecture. Changing this
+  parameter will create a new resource.
+
+* `optional_fields` - (Optional, List) Specifies the optional fields of the data standard template to be activated.
+
+  Changing this parameter will create a new resource.
```
Comment (deer-hang) [inline docs/resources/dataarts_architecture_data_standard_template.md]: This field is not `ForceNew`, so this line should be removed.
Check the other parts.

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	getDataStandardTemplateRespBody, err := utils.FlattenResponse(getDataStandardTemplateResp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataArts Architecture data standard template: %s", err)
+	}
+
+	hasTemplate := utils.PathSearch("data.value.hasTemplate", getDataStandardTemplateRespBody, false).(bool)
+	if !hasTemplate {
+		return nil, fmt.Errorf("error retrieving DataArts Architecture data standard template: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go]:
```suggestion
		return nil, golangsdk.ErrDefault404{}
```

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	rName := "huaweicloud_dataarts_architecture_data_standard_template.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getDataStandardTemplateResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go]:
```suggestion
		PreCheck: func() {
			acceptance.TestAccPreCheck(t)
			acceptance.TestAccPreCheckDataArtsWorkSpaceID(t)
		},
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template.go [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+
+func buildSystemDefaultOrOptionalFields(fields []interface{}, isSystemField bool) []map[string]interface{} {
+	if len(fields) == 0 {
+		return nil
+	}
+
+	rst := make([]map[string]interface{}, len(fields))
+	for i, v := range fields {
+		if raw, ok := v.(map[string]interface{}); ok {
+			rst[i] = map[string]interface{}{
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template.go]: Please use `append` to replace `index`. This line of code may cause unexpected problems when `ok` is false.
Please check the other parts.

File: docs/resources/dataarts_architecture_data_standard_template.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/dataarts_architecture_data_standard_template.md]: done

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go]: done

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template_test.go]: done

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_data_standard_template.go]: done

**Pull request merged**